### PR TITLE
Flow request auth header in as binding data.

### PR DESCRIPTION
**Pull request by MikeStall**

### CODE REVIEW

File: src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs [hunk 1]
```diff
@@ -54,6 +54,9 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
 
         internal class HttpTriggerBinding : ITriggerBinding
         {
+            // Flow the authorization header in as binding data.
+            private const string AuthName = "RequestAuthHeader";
```
Comment (mamaso) [inline src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs]: move this to ScriptConstants.cs

File: src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs [hunk 2]
```diff
@@ -252,6 +256,22 @@ internal static void ApplyBindingData(object target, IDictionary<string, object>
...
+
+                    if (request.Headers.Authorization != null)
+                    {
+                        auth = request.Headers.Authorization.Parameter;
+                    }
+                    else
+                    {
+                        // The (optional) id token from easy auth.
+                        // This can be used in a token exchange.
+                        auth = request.GetHeaderValueOrDefault("X-MS-TOKEN-AAD-ID-TOKEN"); // JWT
```
Comment (mamaso) [inline src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs]: if `Parameter` is null above should we try to pull this header value?  Also, move to script constants.

File: src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs [hunk 3]
```diff
@@ -252,6 +256,22 @@ internal static void ApplyBindingData(object target, IDictionary<string, object>
...
+                    }
+                    bindingData[AuthName] = auth ?? string.Empty;
```
Comment (mamaso) [inline src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs]: General question about binding data: if we set a property in the contract, can we leave that entry in the dictionary undefined? null?

**Comment by fabiocav** [comment]
@mamaso had a chat with @MikeStall about some of the details on this PR. Some things are still in "flux"

**Comment by mathewc** [comment]
This PR doesn't target the dev branch. Recommend closing this PR and opening a new one with the actual changes to be reviewed/merged.

**Comment by MikeStall** [comment]
ok

**Pull request closed without merging**